### ISSUE TITLE
Mask Calculator not working

### ISSUE THREAD

**Issue by Amano47** [Amano47/NeuralHaircut#4]
# calc_masks.py not functioning

There is a problem in the implementation when calculating the masks. It has to do how the old state dict is updated, but also used in the loop to iterate, which results in a mutation.

## Fix

This was also fixed in the issues in the main fork, but here it is anyway.

Simply change the code from Line 156 to 163:  

from this: 
```python
...
156     state_dict = model.state_dict().copy()
157     state_dict_old = torch.load(args.CDGNET_ckpt, map_location='cpu')
158
159-    for key, nkey in zip(state_dict_old.keys(), state_dict.keys()): # delete this line
160         if key != nkey:
161             # remove the 'module.' in the 'key'
162             state_dict[key[7:]] = deepcopy(state_dict_old[key])
...
```

to this:  
```python
...
156     state_dict = model.state_dict().copy()
157     state_dict_old = torch.load(args.CDGNET_ckpt, map_location='cpu')
158
159     state_dict_keys = list(state_dict.keys()) # add this
160     for key, nkey in zip(state_dict_old.keys(), state_dict_keys): # add this
161         if key != nkey:
162             # remove the 'module.' in the 'key'
163             state_dict[key[7:]] = deepcopy(state_dict_old[key])
...
```

and it should work fine.
